producer.go

```diff
@@ -279,6 +285,38 @@ func (opts *ProduceOpts) produce(p *Producer) error {
 	}
 }
 
+func (p *Producer) validateMsg(msg any) ([]byte, error) {
+	sd, err := p.getSchemaDetails()
+	if err != nil {
+		return nil, err
+	}
+
+	if sd.schemaType == "" {
```

Comment:
in case no schema found:
add comment this is the case and validation is not needed